src/test/java/com/rackspace/ceres/app/web/QueryControllerTest.java

```diff
@@ -187,4 +277,21 @@ public void testQueryApiWithNoTenantInHeaderAndParam() {
 
     verifyNoInteractions(queryService);
   }
+
+  @Test
+  public void testQueryApiWithMetricNameAndMetricGroup() {
+    final String metricGroup = RandomStringUtils.randomAlphabetic(5);
+    webTestClient.get()
+        .uri(uriBuilder -> uriBuilder.path("/api/query")
+            .queryParam("metricName", "cpu-idle")
+            .queryParam("metricGroup", metricGroup)
+            .queryParam("tag", "os=linux")
+            .queryParam("start", "1d-ago")
+            .build())
+        .exchange().expectStatus().isBadRequest()
+        .expectBody()
+        .jsonPath("$.status").isEqualTo(400);
+
+    verifyNoInteractions(queryService);
```

Comment:
You should receive 404 invalid request if both name and group are present in the request

Reply:
400 is BAD or Invalid request, 404 is for Not Found

Reply:
yep true

Reply:
I want to see the error code being return in the test here.

Reply:
I think its there `.jsonPath("$.status").isEqualTo(400);`

Reply:
ok I missed that